### PR TITLE
I2C, Janky Input and python2 issues

### PR DESCRIPTION
**Pull request by cnlohr**
Fix I2C.
Fix bad input on Linux.
Fix python2 issues if python2 isn't default accessible or your path isn't set right.

### CODE REVIEW

**Comment by AEFeinstein** [comment]
I tested this, and it indeed fixes the issue. I'm trying to wrap my head around _how_. At a minimum, it does make sense that `memw` comes after where you think it's useful:

> The MEMW instruction causes all memory and cache accesses (loads, stores, acquires, releases, prefetches, and cache operations, but not instruction fetches) **before itself in program order** to access memory before all memory and cache accesses (but not instruction fetches) after. At least one MEMW should be executed in between every load or store to a volatile variable. The Multiprocessor Synchronization Option provides some additional instructions that also affect memory ordering in a more focused fashion. MEMW has broader applications than these other instructions (for example, when reading and writing device registers), but it also may affect performance more than the synchronization instructions.

File: firmware/makefile [hunk 1]
```diff
@@ -265,7 +265,7 @@ all: $(FW_FILE1) $(FW_FILE2) $(ASSETS_FILE)
 
 # To build the two firmware files for flashing, you have to build the main file
 $(FW_FILE1) $(FW_FILE2) : $(FW_FILE0)
-	$(ESP_ROOT)/esptool/esptool.py elf2image $(FW_FILE0)
+	PATH=$(PATH):$(GCC_FOLDER)/bin python2 $(ESP_ROOT)/esptool/esptool.py elf2image $(FW_FILE0)
```
Comment (AEFeinstein) [inline firmware/makefile:268]: `python2` make sense, but why add the xtensa gcc folder to the path?

Comment (AEFeinstein) [reply firmware/makefile:268]: Also might as well replace `python2 $(ESP_ROOT)/esptool/esptool.py` with `$(ESPTOOL_PY)`

Comment (cnlohr) [reply firmware/makefile:268]: It's not natively in my path.  This doesn't seem to be an issue elsewhere but is needed.

Comment (AEFeinstein) [reply firmware/makefile:268]: Ah, adding it to the path is [step 2.3 of the getting started guide](https://github.com/AEFeinstein/Super-2021-Swadge-FW-Sandbox/blob/master/docs/GETTING_STARTED.md), which is why I haven't run into an issue here yet. I'll merge the change, can't hurt.

**Pull request merged**